### ISSUE TITLE
Firefox developer 103, MDM V3.2b4 - Still showing "cannot access internal stroage"

### ISSUE THREAD

**Issue by XxUnkn0wnxX** [jingyu9575/multithreaded-download-manager#180]
![Screen Shot 2022-07-05 at 10 51 49 am](https://user-images.githubusercontent.com/2242880/177229704-f69fecd9-119f-4f73-a06f-49ee4aea03ca.jpg)
is there a way to hide that message? as the instructions for that are a bit dated here: https://github.com/jingyu9575/multithreaded-download-manager/wiki/About-%22Cannot-access-internal-storage%22

there is no "Accept Cookies From Websites" option anymore in firefox.
& the MDM works fine as is. I have that annoying red popup I cannot get rid of, clicking on the white X button does nothing.

**Comment by Remonli** [comment]
Same problem.

**Comment by linsui** [comment]
I reinstalled it and it works again.

**Comment by Evengard** [comment]
103.0b9 dev edition - still the same problem...

**Comment by Onurtag** [comment]
Stable is on 103 now. The same problem also exists there.

**Comment by Anutrix** [comment]
Same issue

**Comment by t0rzz** [comment]
Same issue on Firefox v103.0.2 (64bit)

**Comment by aiprog** [comment]
希望作者能早点修复这个问题。

**Comment by t0rzz** [comment]
@XxUnkn0wnxX please change the title to reflect the actual problem. Internal storage error message appears on Firefox stable too.

**Comment by jingyu9575** [comment]
Sorry for the delay.

This is only a display problem. The extension can be used even if this error shows. (unless "Always use private browsing mode" is on, then the [instructions](https://github.com/jingyu9575/multithreaded-download-manager/wiki/About-%22Cannot-access-internal-storage%22) needs to be done.)

I can fix it, but right now the extension has a more severe problem #179, which can cause downloads to be corrupted. Unfortunately it is a Firefox bug and can only be fixed by updating to Firefox 104 (to be released on 2022-08-23).

If I fix this issue and update the extension now, #179 will still corrupt the files. 

I'm thinking about waiting until 2022-08-23 to update the extension, so that both problems are fixed. Meanwhile, if you are not affected by #179 (not seeing file corruption), please go to the extension options → *Interface* → *Custom CSS*, and add this code:

```css
:root[data-path="panel/panel"] #no-storage-access {
    display: none;
}
```

Then the error will be hidden.

What do you think?

**Comment by Remonli** [comment]
Yes, I also have same problem as #179 , now I use aria2c instead, let's wait for firefox 104.

Anyway, @jingyu9575  thanks for your great job of the past years ,this ext is really very helpful to firefox users.

**Comment by t0rzz** [comment]
L

> Sorry for the delay.
> 
> This is only a display problem. The extension can be used even if this error shows. (unless "Always use private browsing mode" is on, then the [instructions](https://github.com/jingyu9575/multithreaded-download-manager/wiki/About-%22Cannot-access-internal-storage%22) needs to be done.)
> 
> I can fix it, but right now the extension has a more severe problem #179, which can cause downloads to be corrupted. Unfortunately it is a Firefox bug and can only be fixed by updating to Firefox 104 (to be released on 2022-08-23).
> 
> If I fix this issue and update the extension now, #179 will still corrupt the files.
> 
> I'm thinking about waiting until 2022-08-23 to update the extension, so that both problems are fixed. Meanwhile, if you are not affected by #179 (not seeing file corruption), please go to the extension options → _Interface_ → _Custom CSS_, and add this code:
> 
> ```css
> :root[data-path="panel/panel"] #no-storage-access {
>     display: none;
> }
> ```
> 
> Then the error will be hidden.
> 
> What do you think?
> 
> @aiprog Click here for the comment in Chinese 点此展开中文

It’s fine, Firefox 104 will be released in a couple weeks, so we can focus on #179. Storage access display is not a problem, it works fine anyway. Thank you 👍

**Comment by jingyu9575** [comment]
The error has been removed. Please update Firefox and the extension and see if it works.

**Comment by Remonli** [comment]
@jingyu9575  Looks like AMO roll back this ext to 3.2b4 , pls check ?

**Comment by jingyu9575** [comment]
It seems that this version is rejected by the reviewer. Here is the detailed review. I believe it is a mistake as all the mentioned files do not exist in my extension. I've resubmitted and let's see what happens.
```
Details:
This add-on didn&#x27;t pass review because of the following problems:

1)  ‘eval’ is only necessary in rare cases. Please use a different method or explain why eval is required in your add-on.
libs\ToolHelper.js - line 171, 181, 243, 249

In addition, the following is required to complete the review:

1) Your add-on includes a third-party library. Please provide the origin of the exact library version you were using and make sure you are using an exact copy of the original maintainers release version. For more information, refer to https://extensionworkshop.com/documentation/publish/third-party-library-usage/ .
Please add only one version of a library.
libs\layer\mobile\layer.js
libs\layer\layer.js

2) This version contains minified, concatenated or otherwise machine-generated code. Please provide the original sources, together with instructions on how to generate the final XPI. Source code must be provided as an archive and uploaded using the source code upload field, which can be done during submission or on the version page in the developer hub.

Please read through the instructions at https://extensionworkshop.com/documentation/publish/source-code-submission/ .
libs\fetch-filter.min.js
content-script.js

Version(s) affected and disabled:
3.2b5
```

**Comment by jingyu9575** [comment]
Finally version 3.2b7 with this fix passes review. Nothing has been changed except the version number.
